src/test/kotlin/anttracker/ContactTest.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ContactTest.kt
+-----------------
+Revision History
+Rev. 1 - 2024/07/09 Original by M. Baker
+-------------------------------------------
+This file contains the tests for the function
+enterContactInformation from the Contact module.
+---------------------------------
+ */
+import anttracker.contact.enterContactInformation
+import anttracker.db.*
+import io.kotest.core.spec.style.DescribeSpec
+import io.kotest.matchers.collections.shouldContainOnly
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import org.jetbrains.exposed.sql.transactions.transaction
+
+/**
+ * The goal of this test is to confirm that enterContactInformation correctly
+ * creates contacts, either for a singularly created contact, or multiple.
+ * Any contacts created must be inserted into the database.
+ */
+class ContactTest :
+    DescribeSpec({
+        xdescribe("enterContactInformation") {
+            describe("when a contact is created") {
+                it("should be findable in the database") {
+                    // prompts user input to create a contact
+                    val contactCreated = enterContactInformation() ?: throw Exception()
```

Comment:
`enterContactInformation` depends on what the user inputs. Unless we control what the user inputs, this test is not reproducible.

Reply:
Since we only have to check that we can interact with the database, I think we could just test that if we insert N contacts beforehand using a transaction and then fetch N contacts from the database, the fetched contacts should match the inserted contacts.

Reply:
In Russel's lecture, he said creating tests this way were OK, giving an example where a test case opens a prompt to the user to provide information for created entities.

I don't have access to (nor is it exported) the function that directly saves a Contact to the database so this'll be harder to model otherwise, unless we purposefully make a function that saves a contact

Reply:
> Since we only have to check that we can interact with the database, I think we could just test that if we insert N contacts beforehand using a transaction and then fetch N contacts from the database, the fetched contacts should match the inserted contacts.

Hmm, is this really a good test though, as that should be assumed as working on the basis that Exposed is a stable library? Otherwise would this test be checking that the connection is active?

Reply:
With regards to creating the contacts at the beginning of the test, you could do something like this.

```kotlin
                    (0..8).forEach {
                        Contacts.insert {
                            it[name] = "name"
                            it[phoneNumber] = "number"
                            it[email] = "email"
                            it[department] = "department"
                        }
                    }

```

Afterwards, you could check if the fetched contacts from the database match what you inserted.

Reply:
Sure, I'll do this then

Reply:
Could you please clarify more on your first sentence? 

> Hmm, is this really a good test though, as that should be assumed as working on the basis that Exposed is a stable library? Otherwise would this test be checking that the connection is active?

Reply:
Well, building on your suggestion, the test case would test that:

- If Exposed creates N contacts,
- The N contacts exist in the database

Which is only really testing Exposed's logic.

So the test case exists to check that Exposed works as opposed to any logic that AntTracker does

Reply:
Yes. I know it is illogical, but I checked with the professor and he was fine with that being our second unit test.

Reply:
Do you know if he prefers that to a test case which uses user input? Because as I said earlier, it fits something he brought up in lecture

Reply:
Since he did not specify whether the test had to take user input, I believe he will not be biased towards one type of test.

Reply:
Then is there an issue with doing it this way? I need to leave soon and I'm working on my stuff in the doc, otherwise I'll need to make these changes tomorrow

Reply:
No issue at all. I will approve this PR, but I still think we should change this test for the fifth assignment.

Reply:
OK sure, sounds good